module/user-activity/src/commonMain/kotlin/org/cru/godtools/shared/user/activity/model/Badge.kt

```diff
@@ -1,19 +1,28 @@
 package org.cru.godtools.shared.user.activity.model
 
+import com.github.ajalt.colormath.model.RGB
 import org.ccci.gto.support.androidx.annotation.VisibleForTesting
 
-data class Badge(val type: BadgeType, val progress: Int, val target: Int) {
-    enum class BadgeType(@VisibleForTesting internal vararg val variants: Int) {
-        TOOLS_OPENED(1, 5, 10),
-        LESSONS_COMPLETED(1, 5, 10),
-        ARTICLES_OPENED(1, 5, 10),
-        IMAGES_SHARED(1, 5, 10),
-        TIPS_COMPLETED(5, 10, 20),
+data class Badge internal constructor(val type: BadgeType, val progress: Int, val target: Int) {
+    internal companion object {
+        val COLORS_NOT_EARNED = IconColors(base = RGB("#d5d5d5")).alpha(0.38f)
+    }
+
+    enum class BadgeType(
+        @VisibleForTesting internal vararg val variants: Int,
+        internal val colors: IconColors
+    ) {
+        TOOLS_OPENED(1, 5, 10, colors = IconColors(base = RGB("#62CCF3"))),
```

Comment:
the badge variants, so we have a 1, 5 and 10 tools opened badge

Reply:
using constants for them feels like it would just complicate things.

Although how the variants/levels of the badge are defined might change because we are using distinct images for each badge level.

Reply:
I'm also not a huge fan of the default detekt lint checking, I enabled it but for the most part I disagree with a lot of it's checks. I've already disabled it on godtools-android repo several months ago

Reply:
Ah, I see. So there is no badge for intermediates, or does 2-4 tools opened look like badge variant 1 and 6-9 tools opened look like badge variant 5, etc?

Reply:
we actually generate all 3 variants, with each tracking it's own progress.

so if you had opened 3 tools already you would have the following badges:
```kotlin
val badges = listOf(
  Badge(TOOLS_OPENED, progress = 1, target = 1, isEarned = true),
  Badge(TOOLS_OPENED, progress = 3, target = 5, isEarned = false),
  Badge(TOOLS_OPENED, progress = 3, target = 10, isEarned = false)
)
```

Reply:
we coerce "progress" to not exceed the "target" value because badges are earned or not, and the `progress` may be used to display a percent completion UI at some point

Reply:
The UI mockup:
<img width="282" alt="Screen Shot 2023-01-12 at 10 26 18 AM" src="https://user-images.githubusercontent.com/47818/212108337-d85264c4-6ff8-476e-88c4-75add91cab40.png">